---
title: "Differential Gene expression analysis using Deseq2"
author: "Nitin Beesabathuni"
date: "`r Sys.Date()`"
format: 
  gfm:
    toc: true
    toc-depth: 2
    code-fold: true
    theme: cosmo
    fig-format: png
    fig-dpi: 300
---

In this analysis, we walk through a differential gene expression workflow using DESeq2 to study transcriptional changes under hypoxic conditions and to evaluate the effects of OC2 overexpression in LNCaP cells and OC2 knockdown in PC3 cells. The goal is not only to identify differentially expressed genes, but also to demonstrate how to evaluate data quality, interpret model behavior, and assess biological plausibility at each step.

```{r}
#| results: false
#| message: false
#| warning: false
# Load necessary library
library(tidyverse)
library(here)
library(conflicted)
library(DESeq2)
library(ggplot2)
library(ggpubr) 
library(scales) # For readable axis labels
library(gt)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(patchwork)
library(ggplotify)
library(patchwork)
library(org.Hs.eg.db)
library(gridExtra)
library(grid)
library(circlize)
library(ComplexHeatmap)
# Resolve namespace conflicts
conflict_prefer("select", "dplyr")
conflict_prefer("filter", "dplyr")
conflicts_prefer(generics::setdiff)
conflicts_prefer(pheatmap::pheatmap)

#loading functions
source(here("scripts","functions.R"))
```

## 1. Data Import and Metadata Preparation

We begin by importing an annotated gene-level count matrix, using gene symbols as primary identifiers. Sample metadata are constructed to explicitly encode treatment condition and genetic perturbation status, as these variables will later define the design matrix used by DESeq2. Careful metadata preparation at this stage is critical, as all downstream modeling relies on the correctness of these annotations.

```{r}
# Loading cleaned and annotated data for both cell types
data <- read.csv(here("data","analysis_ready_count_data.csv"), row.names = 1, check.names = FALSE)

#Separting data for both cell types into their respective dfs

data_lncap <- data %>%
  select(contains("LNCaP"))

data_pc3 <- data %>%
  select(contains("PC3"))

# 1. Sanitize the column names of your dataframes immediately
# This replaces dashes or illegal characters with dots (e.g., rep-1 becomes rep.1)
colnames(data_lncap) <- make.names(colnames(data_lncap))
colnames(data_pc3) <- make.names(colnames(data_pc3))

# 2. Define the metadata vectors
condition <- c(rep("Normoxia", 4), rep("Hypoxia", 4))
overexpress <- c(rep("Empty_vector", 2), rep("OC2_Overexpress", 2), 
                 rep("Empty_vector", 2), rep("OC2_Overexpress", 2))
knockdown <- c(rep("siCtrl", 2), rep("siOC2", 2), 
               rep("siCtrl", 2), rep("siOC2", 2))

# 3. Create the data frames with explicit row names from the sanitized columns
# We also use stringsAsFactors = TRUE or explicitly wrap in factor() for DESeq2
meta_lncap <- data.frame(
  Treatment = factor(condition, levels = c("Normoxia", "Hypoxia" )),
  Genetic_perturbation = factor(overexpress, levels = c("Empty_vector", "OC2_Overexpress")),
  row.names = colnames(data_lncap) 
)

meta_pc3 <- data.frame(
  Treatment = factor(condition, levels = c("Normoxia", "Hypoxia")),
  Genetic_perturbation = factor(knockdown, levels = c("siCtrl", "siOC2")),
  row.names = colnames(data_pc3)
)

# CRITICAL CHECK: DESeq2 will fail if these are not identical
stopifnot(all(colnames(data_lncap) == rownames(meta_lncap)))
stopifnot(all(colnames(data_pc3) == rownames(meta_pc3)))
```

## 2. Data QC for any technical artifacts

Before performing differential expression analysis, it is essential to assess whether technical artifacts or outlier samples are present. While replicate correlations at the count level provide a useful first check, high-dimensional methods such as principal component analysis (PCA) and hierarchical clustering offer deeper insight into global sample relationships.

```{r}
#| warning: false
#| fig-width: 6
#| fig-height: 4
#| fig-dpi: 300
##defining deseq2 object
dds_pc3<- DESeqDataSetFromMatrix(countData = data_pc3,
                              colData = meta_pc3,
                              design = ~Treatment*Genetic_perturbation)   
dds_lncap<- DESeqDataSetFromMatrix(countData = data_lncap,
                              colData = meta_lncap,
                              design = ~Treatment*Genetic_perturbation) 
# For PC3
pc3_QC<-QC_pca_heatmap(dds_pc3, meta_pc3,cell_line_name = "PC3")

# For LNCaP
lncap_QC<-QC_pca_heatmap(dds_lncap, meta_lncap,cell_line_name = "LNCaP")
```

```{r}
#| warning: false
#| fig-width: 10
#| fig-height: 3
#| fig-dpi: 300
#PCA for both the cell lines across two treatments and genetic perturbations. 
lncap_QC$pca+ pc3_QC$pca 
```

***Figure 1:** Principal component analysis for LNCaP and PC3 samples. Genetic perturbation is displayed by color and Treatment condition is displayed by different symbols.*

For LNCaP samples, PCA reveals four well-separated clusters corresponding to the combination of treatment condition and genetic perturbation. The first principal component (PC1), which explains approximately 56% of the total variance, primarily captures treatment effects. The second principal component (PC2), explaining approximately 37% of the variance, reflects the impact of OC2 overexpression. The absence of isolated samples or unexpected clustering patterns suggests that these data are free of major technical artifacts and suitable for downstream modeling.

PC3 samples exhibit a different structure. PCA shows a strong separation between siCtrl and siOC2 conditions; however, within the siOC2 group, samples further separate into two distinct clusters- not based on the treatment. This suggesting the presence of an additional latent variable. To investigate this further, lets examine sample correlation using heatmaps.

```{r}
#| fig-width: 8
#| fig-height: 5
#| fig-dpi: 300
#| warning: false

#heatmap for pc3 cells
pc3_QC$heatmap
```

***Figure 2:** Heatmap of correlation matrix between samples for PC3 cell line. Correlation is measured using pearson correlation metric.*

Based on the heatmap, we see another latent variable- the siRNA used. We can see the siRNA specific samples cluster more strongly together regardless of the treatment. If treatment effects dominate global expression variance, we would expect the siRNA1 and siRNA2 to cluster together based on the treatment. However, we see a stronger correlation between the same siRNA's used for knockdown of OC2 regardless of the treatment. This could perhaps be a result of knockdown of OC2 which rescues some aspects of hypoxia driven cellular changes (not fully). However, we also see the siRNA's under normoxia cluster separately so there could be some off target activity of the specific siRNA used to knockdown OC2.

```{r}
#| warning: false
#| fig-width: 8
#| fig-height: 5
#| fig-dpi: 300
#heatmap for lncap cells
lncap_QC$heatmap
```

***Figure 3:** Heatmap of correlation matrix between samples for LNCaP cell line. Correlation is measured using pearson correlation metric.*

For LNCaP cells, the replicates cluster together based on the treatment and genetic perturbation as expected. Next, lets perform differential expression analysis using Deseq2 to ask various questions.

## 3. Differential expression analysis using Deseq2

With data quality assessed, we proceed to differential expression analysis using DESeq2. Reference levels are set such that normoxia and control genetic perturbations (empty vector or siCtrl) serve as baseline conditions. Calling the `DESeq()` function initiates a sequence of modeling steps, which are outlined below to clarify how DESeq2 arrives at its final statistical tests.

```{r}
#| warning: false
#releveling to set reference for treatment and perturbation
dds_pc3$Treatment <- relevel(dds_pc3$Treatment, ref = "Normoxia")
dds_pc3$Genetic_perturbation <- relevel(dds_pc3$Genetic_perturbation, ref = "siCtrl")
  
dds_lncap$Treatment <- relevel(dds_lncap$Treatment, ref = "Normoxia")
dds_lncap$Genetic_perturbation <- relevel(dds_lncap$Genetic_perturbation, ref = "Empty_vector")

dds_lncap <- DESeq(dds_lncap)
dds_pc3<-DESeq(dds_pc3)
```

You can see various steps that are performed by this function. Briefly, the DESeq function performs the following steps-

1\) **estimating size factors**- Estimate sample-specific size factors using the median-of-ratios method to normalize for sequencing depth and composition bias.

2\) **estimating dispersions/gene-wise dispersion estimates**- Estimate gene-wise dispersion parameters under a negative binomial model.

3\) **mean-dispersion relationship**- Fit a mean–dispersion trend across genes.

4\) **final dispersion estimates**-Calculate the final dispersion estimates using shrinkage toward the fitted trend.

5\) **fitting model and testing**- Fit a negative binomial generalized linear model using the specified design matrix and perform hypothesis testing (Using Wald test here) to identify differentially expressed genes.

Now, lets examine the mean-dispersion plot to check how well our assumptions hold and

```{r}
#| warning: false
#| fig-width: 12
#| fig-height: 5
#| fig-dpi: 300
# Set the graphics device to 1 row and 2 columns
par(mfrow = c(1, 2))

# Plot for LNCaP with a custom title
plotDispEsts(dds_lncap, main = "Dispersion Estimates: LNCaP")

# Plot for PC3 with a custom title
plotDispEsts(dds_pc3, main = "Dispersion Estimates: PC3")
```

***Figure 4:** Dispersion estimates for both cell lines using Deseq2 package. Each dot represents a gene.*

The fit looks good. We can see the fitted (Red) smoothly follows the data points and shrinkage in dispersion (blue dots) towards fitted curve nicely. Also, notice a nice downward trend in dispersion as mean increases. We can also observe some final dispersion and estimated values perfectly overlap, these are genes that are not shrunk towards the fitted line.

```{r}
# Reset the graphics device to default (1x1)
par(mfrow = c(1, 1))

#lets understand the genes that are differential expressed under hypoxia condition. 
# PC3: Hypoxia vs Normoxia (within siCtrl)
res_pc3_hypoxia <- results(dds_pc3, 
                           contrast = c("Treatment", "Hypoxia", "Normoxia"))

# LNCaP: Hypoxia vs Normoxia (within empty vector)
res_lncap_hypoxia <- results(dds_lncap, 
                             contrast = c("Treatment", "Hypoxia", "Normoxia"))

```

Prior to examining individual differentially expressed genes (DEGs), lets examine the relationship between normalized counts and estimated fold changes to ensure that model estimates behave sensibly across the expression range.

```{r}
#| warning: false
#| fig-width: 10
#| fig-height: 5
#| fig-dpi: 300
# 1. Prepare the grid (1 row, 2 columns)
par(mfrow = c(1, 2))

# 2. Call the plots directly
plotMA(res_lncap_hypoxia, ylim = c(-4, 4), alpha = 0.05,main = "LNCaP: Hypoxia vs Normoxia (Empty Vector)")
plotMA(res_pc3_hypoxia, ylim = c(-4, 4), alpha = 0.05,main = "PC3: Hypoxia vs Normoxia (siCtrls)")

# 3. Reset to default (1x1)
par(mfrow = c(1, 1))
```

**Figure 5: MA plots illustrating global transcriptional changes in response to hypoxia.** Each data point represents an individual gene. Differentially expressed genes (DEGs) between normoxia and hypoxia—identified in control samples (Empty Vector for LNCaP and siCtrl for PC3)—are highlighted in blue based on a significance threshold of *padj* \< 0.05.

Overall, we identify a larger number of DEGs in LNCaP cells compared to PC3 cells. This difference may reflect biological differences between the cell lines, differences in variability across samples, or latent technical factors.

## 4. Differentially expressed genes under hypoxia condition

```{r}
#| message: false
#| warning: false
#| fig-width: 20
#| fig-height: 8
#| fig-dpi: 300

# Generate plots for both cell lines
v_pc3 <- plot_volcano(res_pc3_hypoxia, "PC3: Hypoxia vs Normoxia (siCtrl)",p_clip = 1e-50)
v_lncap <- plot_volcano(res_lncap_hypoxia, "LNCaP: Hypoxia vs Normoxia (Empty vector)",p_clip = 1e-50)

# Display side-by-side
 v_lncap+v_pc3
```

***Figure 6:** Volcano plot displaying DEGs between normoxia and hypoxia conditions for empty vector and siCtrl for LNCaP and PC3 Cells, respectively. Fold change of 2 and with padj of 0.05 were used as thresholds.*

We can see a good number of genes that are differentially expressed. For PC3 samples, we can see some well known genes regulated under hypoxic conditions such as ANGPTL4 and NDRG1. Lets perform pathway enrichment for these DEGs to understand the pathways regulated and to confirm that known biology is recoverd.

```{r}
#| message: false
#| warning: false
#| fig-width: 17.5
#| fig-height: 5
#| fig-dpi: 300
# extract entrezids from gene symbols
get_sig_genes <- function(res) {
  # Filter for padj < 0.05 and |LFC| > 1
  sig <- subset(res, padj < 0.05 & abs(log2FoldChange) > 1)
  
  # Convert Gene Symbols (rownames) to Entrez IDs
  ids <- bitr(rownames(sig), fromType = "SYMBOL", 
              toType = "ENTREZID", OrgDb = org.Hs.eg.db)
  return(ids$ENTREZID)
}

genes_pc3 <- get_sig_genes(res_pc3_hypoxia)
genes_lncap <- get_sig_genes(res_lncap_hypoxia)

pc3_kegg<-plot_kegg_pathways(genes_pc3, "PC3: Hypoxia Regulated Pathways", p_cutoff = 0.05, max_pathways = 10)

# For LNCaP
lncap_kegg<-plot_kegg_pathways(genes_lncap, "LNCaP: Hypoxia Regulated Pathways", p_cutoff = 0.05, max_pathways = 10)

lncap_kegg+pc3_kegg
```

***Figure 7: Functional enrichment of hypoxia-responsive genes.** Plots displaying KEGG pathway enrichment analysis of DEGs between normoxia and hypoxia. Comparisons were performed in control samples: Empty Vector for LNCaP and siCtrl for PC3. Statistical significance is represented as -log10(padj).*

We can observe Glycolysis and HIF-1 signaling pathways pop up which are some of the well known pathways regulated under under hypoxia conditions. This provides additional validation for the analysis.

## 5. Role of ONECUT2 under hypoxia conditions

To understand the role of OC2 gene and the mechanism of action under hypoxia conditions, lets first analyze PC3 samples where OC2 is knocked down. Lets first sanity check if we observe a decrease in PC3 transcripts

```{r}
#| message: false
#| warning: false
#| fig-width: 4
#| fig-height: 2
#| fig-dpi: 300
# 1. Extract normalized counts for a specific gene (e.g., AAMP)
gene_name <- "ONECUT2"
plot_data <- plotCounts(dds_pc3, gene = gene_name, intgroup = c("Treatment", "Genetic_perturbation"), returnData = TRUE)

# 2. Add a column for Sample Names (the x-axis)
plot_data$Sample <- rownames(plot_data)

ggplot(plot_data, aes(x = Genetic_perturbation, y = count, fill = Genetic_perturbation)) +
  # This groups Normoxia and Hypoxia into separate, adjacent panels
  facet_wrap(~Treatment) + 
  # Using 'summary' to show the mean height of the replicates
  geom_bar(stat = "summary", fun = "mean", color = "black", width = 0.7) +
  # Adding individual points to show the actual replicate values (important for QC)
  geom_point(position = position_jitter(width = 0.1), color = "black") +
  theme_minimal() +
  labs(
    x = "Genetic Perturbation",
    y = "Normalized OC2 Counts",
    fill = "Group"
  ) +
  theme(
    legend.position = "none",
    strip.text = element_text(face = "bold", size = 6), 
    axis.title = element_text(size = 7),
    axis.text = element_text(size = 6),
    panel.spacing = unit(2, "lines") # Adds space between the two panels
  )
```

***Figure 8:*** Confirmation of reduction in *ONECUT2* transcripts following siRNA knockdown across both conditions. *Normalized count shown on y-axis. Rep1/ rep2 for siCtrl and two independent guides for siOC2 as shown as individual datapoints.*

We can notice a strong reduction (5X) in siOC2 transcripts compared to siCtrl in both treatment groups. Lets analyze the genes and pathways modulated by siOC2 knockdown.

```{r}
res_OC2_normoxia <- results(dds_pc3, 
                           contrast = c("Genetic_perturbation", "siOC2", "siCtrl"))

# Effect of siOC2 compared to siCtrl specifically in Hypoxia
res_OC2_Hypoxia <- results(dds_pc3, 
                             contrast = list(c("Genetic_perturbation_siOC2_vs_siCtrl", 
                                               "TreatmentHypoxia.Genetic_perturbationsiOC2")))

```

```{r}
#| warning: false
#| fig-width: 10
#| fig-height: 5
#| fig-dpi: 300
# 1. Prepare the grid (1 row, 2 columns)
par(mfrow = c(1, 2))

# 2. Call the plots directly
plotMA(res_OC2_normoxia, ylim = c(-4, 4), alpha = 0.05,main = "Normoxia: siCtrl vs siOC2")
plotMA(res_OC2_Hypoxia, ylim = c(-4, 4), alpha = 0.05,main = "Hypoxia: siCtrl vs siOC2")

# 3. Reset to default (1x1)
par(mfrow = c(1, 1))
```

***Figure 9:** MA plots for genes regulated by siOC2 knockdown under normoxia and hypoxia conditions.*

Lets visualize the DEGs using volcano plots

```{r}
#| message: false
#| warning: false
#| fig-width: 15
#| fig-height: 8
#| fig-dpi: 300

# Generate plots for both cell lines
norm_sioc2 <- plot_volcano(res_OC2_normoxia, "PC3-Normoxia: siCtrl vs siOC2")
hypox_sioc2 <- plot_volcano(res_OC2_Hypoxia, "PC3-Hypoxia: siCtrl vs siOC2")

# Display side-by-side
norm_sioc2+hypox_sioc2
```

***Figure 10:** Volcano plots displaying DEGs between siCtrl and siOC2 under normoxia and hypoxia treatments for PC3 samples.*

We can observe some conistent genes that change in the same direction. For example, CLDN1 is upregulated while NOX5 is down regulated under both treatment conditions. Lets also perform a sanity check to confirm the results we are seeing are consistent with normalized counts.

```{r}
#| message: false
#| warning: false
#| fig-width: 12
#| fig-height: 6
#| fig-dpi: 300

# --- 1. PREPARE DATA ---
norm_counts <- counts(dds_pc3, normalized = TRUE)
norm_samples <- rownames(colData(dds_pc3)[colData(dds_pc3)$Treatment == "Normoxia", ])
hypo_samples <- rownames(colData(dds_pc3)[colData(dds_pc3)$Treatment == "Hypoxia", ])

sig_genes_norm <- rownames(subset(res_OC2_normoxia, padj < 0.05 & abs(log2FoldChange) > 1))
sig_genes_hypo <- rownames(subset(res_OC2_Hypoxia, padj < 0.05 & abs(log2FoldChange) > 1))

# --- 2. GENERATE HEATMAPS AS OBJECTS ---
#normoxia heatmap
p1 <- pheatmap::pheatmap(t(scale(t(norm_counts[sig_genes_norm, norm_samples]))), 
                         cluster_cols = TRUE, cluster_rows = TRUE, 
                         show_rownames = FALSE, angle_col = 45, fontsize_col = 10,
                         main = paste("Normoxia:", length(sig_genes_norm), "DEGs"),
                         color = colorRampPalette(c("royalblue4", "white", "firebrick3"))(100),
                         border_color = NA, silent = TRUE)$gtable

#hypoxia heatmap
p2 <- pheatmap::pheatmap(t(scale(t(norm_counts[sig_genes_hypo, hypo_samples]))), 
                         cluster_cols = TRUE, cluster_rows = TRUE, 
                         show_rownames = FALSE, angle_col = 45, fontsize_col = 10,
                         main = paste("Hypoxia:", length(sig_genes_hypo), "DEGs"),
                         color = colorRampPalette(c("royalblue4", "white", "firebrick3"))(100),
                         border_color = NA, silent = TRUE)$gtable

# --- 3. ARRANGE BOTH---
grid.arrange(p1, p2, ncol = 2)
```

Figure 11: **Heatmap of normalized counts for differentially expressed genes (DEGs) following siOC2 knockdown for PC3 samples.**

The heatmaps generated from normalized counts align with the expected expression patterns, thereby confirming the reliability of the identified DEGs and demonstrating that the results are not due to technical artifacts.

### The Rescue Signature

Lets look at the overlap between hypoxia specific genes and genes regulated by siOC2 knockdown. Particularly, interested in the genes "rescued" in the opposite directions.

```{r}
#| message: false
#| warning: false
#| fig-width: 12
#| fig-height: 8 
#| fig-dpi: 300

# --- 1. IDENTIFY ALL RESCUED GENES IN OPPOSITE DIRECTIONS---
hypo_up <- rownames(subset(res_pc3_hypoxia, padj < 0.05 & log2FoldChange > 1))
oc2_down <- rownames(subset(res_OC2_Hypoxia, padj < 0.05 & log2FoldChange < -1))
rescued_up_down <- base::intersect(hypo_up, oc2_down)

hypo_down <- rownames(subset(res_pc3_hypoxia, padj < 0.05 & log2FoldChange < -1))
oc2_up <- rownames(subset(res_OC2_Hypoxia, padj < 0.05 & log2FoldChange > 1))
rescued_down_up <- base::intersect(hypo_down, oc2_up)

all_rescued <- c(rescued_up_down, rescued_down_up)

# --- 2. PREPARE & CLEAN DATA ---
norm_counts <- counts(dds_pc3, normalized = TRUE)
sample_order <- c(
  rownames(colData(dds_pc3)[colData(dds_pc3)$Treatment == "Normoxia" & colData(dds_pc3)$Genetic_perturbation == "siCtrl", ]),
  rownames(colData(dds_pc3)[colData(dds_pc3)$Treatment == "Normoxia" & colData(dds_pc3)$Genetic_perturbation == "siOC2", ]),
  rownames(colData(dds_pc3)[colData(dds_pc3)$Treatment == "Hypoxia" & colData(dds_pc3)$Genetic_perturbation == "siCtrl", ]),
  rownames(colData(dds_pc3)[colData(dds_pc3)$Treatment == "Hypoxia" & colData(dds_pc3)$Genetic_perturbation == "siOC2", ])
)

mat_rescued <- norm_counts[all_rescued, sample_order]
# Remove zero-variance rows to prevent scaling errors
#mat_rescued <- mat_rescued[apply(mat_rescued, 1, var) > 0, ]
mat_scaled <- t(scale(t(mat_rescued)))

# Clip for contrast
mat_scaled[mat_scaled > 2] <- 2
mat_scaled[mat_scaled < -2] <- -2

# Recalculate splits for surviving genes
final_genes <- rownames(mat_scaled)
split_factor <- ifelse(final_genes %in% rescued_up_down, "Inhibited Induction", "Reversed Suppression")

# --- 3. PLOT ---
col_fun = colorRamp2(c(-2, 0, 2), c("#2166AC", "#FFFFFF", "#B2182B"))

ann <- data.frame(
  Treatment = colData(dds_pc3)[sample_order, "Treatment"],
  Knockdown = colData(dds_pc3)[sample_order, "Genetic_perturbation"]
)
col_ann <- list(
  Treatment = c("Normoxia" = "#99CCFF", "Hypoxia" = "#003366"),
  Knockdown = c("siCtrl" = "grey80", "siOC2" = "purple")
)

Heatmap(mat_scaled, 
        name = "Z-score",
        column_title = paste("Rescue Gene Signature:", nrow(mat_scaled), "Genes"),
        col = col_fun,
        row_split = split_factor,
        show_row_names = TRUE, 
        row_names_side = "right",
        row_names_gp = gpar(fontsize = 8, fontface = "italic"), 
        show_row_dend = FALSE,
        
        cluster_columns = FALSE, 
        column_names_rot = -60,
        top_annotation = HeatmapAnnotation(df = ann, col = col_ann),
        border = TRUE,
        rect_gp = gpar(col = "white", lwd = 0.1))
```

**Figure 12: Identification of the OC2-mediated "Rescue Signature" under hypoxic conditions.** Genes regulated by hypoxia were first identified by comparing siCtrl Normoxia and siCtrl Hypoxia samples. Rescued genes were defined as those exhibiting a minimum 2-fold change in opposite directions between the hypoxia response and the siOC2 knockdown, with a significance threshold of *padj* \< 0.05.

## 6. Summary and Conclusions

In this analysis, we conducted a differential gene expression analysis using the **DESeq2** package for PC3 and LNCaP cell lines under hypoxic conditions.

We implemented several quality control (QC) measures to evaluate data integrity and identify potential technical artifacts.

We performed differential gene analysis using Deseq2 packaged and identified differentially expressed genes (DEGs) for PC3 and LNCaP cells under hypoxia conditions. We validated these findings through pathway enrichment to confirm the biological accuracy of our results.

Furthermore, we characterized the transcriptional role of ONECUT2 (OC2) in PC3 cells by analyzing the effects of siRNA-mediated knockdown. This allowed us to pinpoint a specific 'Rescue Signature'—genes modulated by OC2 knockdown specifically under hypoxic conditions.
